### ISSUE TITLE
Error: Attempting to duplicate an element in an unsupported site

### ISSUE THREAD

**Issue by jpeacock** [engram-design/SuperTable#357]
### Description
Updated from 2.1.20 to latest and there's a stalled/failed task. 

Task is "Applying new propagation method to Super Table blocks". 
Error is "Attempting to duplicate an element in an unsupported site"

Site is a multi-site installation with some SuperTable blocks in a single Matrix block. 

### Steps to reproduce

1. Clicking "Retry" results in a failed task immediately. 

### Additional info

- Plugin version: 2.4.9
- Craft version: 3.4.20
<img width="1210" alt="Screen Shot 2020-05-20 at 11 26 19 AM" src="https://user-images.githubusercontent.com/223784/82473153-c8049b00-9a8e-11ea-8bec-eed316b5b92c.png">

**Comment by richardegil** [comment]
@jpeacock did you find a solution to this? I think I might be having the same issue.

**Comment by jpeacock** [comment]
@richardegil not yet

**Comment by engram-design** [comment]
@jpeacock Haven't come across this, which is odd. I know Neo is having issues with this (and might be the issue for @richardegil). I assume you don't have Neo installed @jpeacock ?

**Comment by engram-design** [comment]
In addition, if you can have a look for this error in your `queue.log` file and send through the output of that? That'll hopefully give me a bit more context into the issue...

**Comment by jpeacock** [comment]
@engram-design nope, Neo is not installed. 

Here's a snippet from the log from that timestamp in the queue - hoping it helps
[queue.log.txt](https://github.com/verbb/super-table/files/4705959/queue.log.txt)

**Comment by engram-design** [comment]
Thanks for the info. I don't suppose its easy enough to tell which Matrix + Super Table field this might've been referring to? If so, what are the translation/propagation settings for both the Matrix and Super Table fields?

**Comment by jpeacock** [comment]
Is there a way to find that info out fairly easily which specific field instance is causing the issue? The install is spread across 4 sites with a main Matrix block that houses a bunch of HTML components, some with translations and some not. There are a bunch of Super Table blocks in that Matrix field.

**Comment by arthurdewolf** [comment]
We have this same issue with multiple fields on a multi-site:

![image](https://user-images.githubusercontent.com/5043193/87441919-aa047400-c5c1-11ea-9a53-396cb612c78e.png)

Each failed job has the error `Attempting to duplicate an element in an unsupported site`.

![image](https://user-images.githubusercontent.com/5043193/87442207-ff408580-c5c1-11ea-8a6a-f95b0150d7d1.png)

All Matrix and Super Table fields have Propagation Method set to "Save blocks to other sites in the same site group".

**Comment by engram-design** [comment]
@arthurdewolf Was this a similar upgrade from a previous version (of Craft or Super Table)? If so, can you note the versions?

Investigating - I can't reproduce as-is, so will need to try upgrading from previous persons to see if its a migration specific thing.

**Comment by engram-design** [comment]
@brandonkelly has done some digging on this, and thinks the changes here - https://github.com/craftcms/cms/commit/30c43180b78270c2c13e103ed9a5951f58353e5e - might improve things.

Somehow, the data for Matrix/Super Table blocks has gotten messed up, by it being in the wrong site in the first place. However, Craft hard-errors this, when it probably doesn't have to, hence the failed queue tasks.

In addition, I've just added some better handling with the re-saving fields utility. It now better handles Matrix + Super Table combinations.

To get the fix early, change your `verbb/super-table` requirement in composer.json to:

```
"require": {
  "verbb/super-table": "dev-craft-3 as 2.5.1",
  "...": "..."
}
```

Then run `composer update`

**Comment by arthurdewolf** [comment]
Thank you @engram-design! These Craft and Super Table changes seem to fix it for us.

**Comment by jpeacock** [comment]
@engram-design it now lets me restart the task, shows progress for a bit, then errors out after a while. On refresh, the total it's counting towards has increased. 
<img width="478" alt="Screen Shot 2020-07-20 at 11 12 12 AM" src="https://user-images.githubusercontent.com/223784/87960561-17108180-ca7a-11ea-8f8c-f98d3f2423de.png">
<img width="636" alt="Screen Shot 2020-07-20 at 11 12 20 AM" src="https://user-images.githubusercontent.com/223784/87960563-17a91800-ca7a-11ea-9c2f-5c35e5c7bea5.png">
<img width="453" alt="Screen Shot 2020-07-20 at 11 12 33 AM" src="https://user-images.githubusercontent.com/223784/87960564-1841ae80-ca7a-11ea-9348-bc51bf1e3e73.png">

**Comment by engram-design** [comment]
@jpeacock Is "after a while" longer than 5 minutes? Wondering if you're hitting the timeout. If you're on the latest Craft branch, you shouldn't get that error at all.

**Comment by jpeacock** [comment]
No, it's anywhere from 8-10 seconds to 20. Hasn't been more than 20 yet. Craft version 3.4.28.1

**Comment by engram-design** [comment]
Wondering if you can cancel those jobs, then run the Super Table re-save field command to see if they crop up again?

**Comment by jpeacock** [comment]
@engram-design how do I run the re-save field? should I just resave the entries via the Craft CLI?

**Comment by engram-design** [comment]
@jpeacock Nope, re-saving entries is a different thing. There should be some info in Settings > Super Table `admin/super-table/settings`. There's are-save field button that's a quick way of saving all your Super Table fields.

The propagation queue tasks shouldn't actually fire unless something has changed with the field settings. As such, this is a good test to ensure each Super Table has the correct settings. There's better handing for combinations with Matrix fields, so just make sure you're on the latest as per my previous `dev-craft-3` code.

**Comment by jpeacock** [comment]
@engram-design ah, gotcha. I never knew that was there. I ran the re-save task after deleting the other error'd out item in the queue and the problem seems to be resolved. Thanks!

**Comment by engram-design** [comment]
Fixed in [2.5.2](https://github.com/verbb/super-table/releases/tag/2.5.2)
